### ISSUE TITLE
confidence score inconsistency with original paper

### ISSUE THREAD

**Issue by harryxu-yscz** [experiencor/basic-yolo-keras#8]
This is a question related with the content of the yolov2 paper. In paper, the predicted confidence pred_c is the product of class probability and IOU:

Pr(object)*IOU(box,object) =  sigmoid(pred_c)

However, in the ```interpret_netout``` function, I saw you do: Pr(object)=Pr(object)*sigmoid(pred_c), where Pr(object) uses softmax(pred_prob). While it makes sense to do so, I don't see the relationship between that and the above equation in paper.. and I don't know if this is the approach used in darknet?

Could you help me understand your approach here? Thanks!

**Comment by Mirorrn** [comment]
Search for  "class-specific confidence scores" in the original yolo v1 paper: https://pjreddie.com/media/files/papers/yolo_1.pdf

**Comment by experiencor** [comment]
@harryxu-yscz, I didn't notice this point until you asked the question. It turns out that I didn't follow the YOLOv1 paper closely regarding this matter of confidence score. Specifically, I completely ignore the IOU business when dealing with confidence. So Pr(object) is directly the confidence score (or objectness score). I will revise the code with regard to this matter.

In the `interpret_netout`, I did: Pr(class) = P(class/object) * Pr(object), not the formular that you wrote.

I hope that it clears the confusion.

**Comment by harryxu-yscz** [comment]
Guys thanks for the engagements.

@experiencor you addressed my questions clearer than I did. And thanks for pointing out the details in ```interpret_netout```

**Comment by harryxu-yscz** [comment]
Did we solve this problem in the code?

**Comment by experiencor** [comment]
@harryxu-yscz, I have fixed this code but there are still some discrepancies with the original loss function. I will update the code when I fix all of them. Stay tuned.

**Comment by mrqorib** [comment]
@experiencor Hi experioncor, I just want to ask, is this fixed yet?

**Comment by experiencor** [comment]
@mrqorib Yes. This issue was fixed in the latest code.
